common/djangoapps/nodebb/signals/handlers.py

```diff
@@ -92,19 +93,23 @@ def create_user_on_nodebb(sender, instance, created, **kwargs):
         return status_code
 
 
-@receiver(pre_save, sender=User, dispatch_uid='activate_user_on_nodebb')
-def activate_user_on_nodebb(sender, instance, **kwargs):
+@receiver(pre_save, sender=User, dispatch_uid='activate_deactivate_user_on_nodebb')
+def activate_deactivate_user_on_nodebb(sender, instance, **kwargs):
     """
-    Activate a user on nodebb whenever user become active on edx platform
+    Activate or Deactivate a user on nodebb whenever user's active state changes on edx platform
     """
-    if instance.is_active and User.objects.filter(pk=instance.pk, is_active=False).exists():
+    current_user_obj = User.objects.filter(pk=instance.pk).first()
```

Comment:
Why we are calling .first method here? I think it's not possible that we have user duplication

Reply:
we are getting based on primary key. So It will be just one entry or None.

Reply:
I have done it so that we don't have to use try except blocks.

Reply:
I think this we should not use .first as we are not using current_user_obj anywhere except in conditions. We don't need the first object, we just need to check if user exists. That can achieved just using User.objects.filter(pk=instance.pk)

Reply:
we need it's is_active field